### ISSUE TITLE
Add a sinusoidal embedding layer

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#26]
Along with a learned positional embedding (https://github.com/keras-team/keras-nlp/issues/23), we should add a fixed sin/cos embedding as described in [Attention is all you need](https://arxiv.org/pdf/1706.03762.pdf).

**Comment by amantayal44** [comment]
I am familiar with transformer and TensorFlow, interested in taking up this issue.

**Comment by mattdangerw** [comment]
@amantayal44 that would be great!

@hertschuh is currently working on #23 which should be up for review shortly. It would probably be good to wait for us to at least discuss the overall design on that embedding layer before starting on this, so we can align on naming and the like. Then the code for that layer could be a good launching off point for this.

**Comment by amantayal44** [comment]
> OK, I think we have a clear enough path forward here. Let's name this `SinusoidalPositionalEmbedding` and include a `max_length` argument (though we may still tweak these). Mimic the inputs and outputs of #47, you should be able to swap one for the other.
> 
> A possible starting point for code for this is [here](https://github.com/tensorflow/models/blob/e0ed507504b4efa6780e244677ded563b36b0ee7/official/nlp/modeling/layers/position_embedding.py#L98).
> 
> @amantayal44 would you still like to work on this?

yes sure

**Comment by mattdangerw** [comment]
Thank you!

re 1) good catch, let's stick to the original paper details when possible.

re 2) also good point, that's my bad. We can drop the max_length argument.

A couple other points from https://github.com/keras-team/keras-nlp/issues/23 
 - The learned position embedding layer will be assume the `-2` axis is the sequence axis. We can do the same here for now (and possibly open this as a config option in the future).
 -  Also, the naming discussion continues over on [this thread](https://github.com/keras-team/keras-nlp/pull/47/files/3d2bf5c5ccd0791ef726e23acbeba8d51e5df3cb#r828478237). Tentative new name `SinePositionEncoding`, we will see if that sticks 😁
